packages/core/admin/admin/src/content-manager/components/RelationInput/RelationInput.js

```diff
@@ -34,20 +51,54 @@ const RelationInput = ({
   name,
   label,
   labelLoadMore,
-  listHeight,
   loadingMessage,
-  relations,
   onRelationAdd,
   onRelationLoadMore,
-  onSearchClose,
-  onSearchOpen,
   onRelationRemove,
+  onSearchClose,
   onSearchNextPage,
+  onSearchOpen,
   onSearch,
   placeholder,
   publicationStateTranslations,
+  relations,
   searchResults,
 }) => {
+  const listRef = useRef();
+  const [overflow, setOverflow] = useState('bottom');
+
+  // To clean (number relations + numberOfRelationsToDisplay as prop?)
+  const totalNumberOfRelations = relations.data?.pages[0]?.length;
```

Comment:
```suggestion
  const totalNumberOfRelations = relations.data?.pages.flat().length;
```

It might also make sense to store the result of `relations.data?.pages.flat()` before and re-use it later, so that we don't have to do the same computation twice.

You also need to make sure, that it defaults: 1) either to `0` instead of `undefined` because you are computing below, which may lead to `NaN` or 2) you need to add a truthy check below.